File: app/src/main/java/ruan/com/Net/MqttManager/Interface/MqttCallback.java
package ruan.com.Net.MqttManager.Interface;

import ruan.com.Net.HttpManager.HttpResponse;

/**
 * Created by 19820 on 2018/1/8.
 */

public interface MqttCallback {


    interface Response{
        /**
         * 请求成功
         * @param requestCode       请求标识
         * @param message        请求返回的数据
         */
        void onSuccess(String requestCode, String message);

        /**
         * 请求失败
         * @param requestCode       请求标识
         * @param throwable         请求失败的对象
         */
        void onError(String requestCode, Throwable throwable);

    }
}

File: app/src/main/java/ruan/com/Net/MqttManager/Interface/MqttCallback.kt
package ruan.com.Net.MqttManager.Interface

/**
 * Created by Administrator on 2018/5/4.
 */
interface MqttCallback {

    interface Response {
        /**
         * 请求成功
         * @param topic       请求标识
         * @param message        请求返回的数据
         */
        fun onSuccess(topic: String?, message: String?)

        /**
         * 请求失败
         * @param topic       请求标识
         * @param throwable         请求失败的对象
         */
        fun onError(topic: String?, throwable: Throwable?)

    }
}